### PR TITLE
Change fuzzy match search scenario so that it tests supported behaviour

### PR DESCRIPTION
**Pull request by davidhamill1-nhs**
## Summary
* Routine Change - Searching on historic date of births was never a supported feature in PDS FHIR nor in PDS HL7. This test worked because of a bug in PDS *not* filtering out stale indexes when searching on the inverted index. With the fix implemented in SpineII under SPINEDEM-3846, this test will fail, as it should have done from the beginning. It is now changed to a field on which one can search historic values, family name.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js [hunk 1]
```diff
@@ -149,14 +150,20 @@ if (request.pathMatches('/Patient') && request.get) {
 
   if (validateHeaders(request) && validateQueryParams(request)) {
     if (fuzzyMatch) {
-      if (family === 'Garton' && (given[0]) === 'Bill' && birthDate[0] === '1946-06-23') {
+      if (family === 'Blogs' && given[0] === 'Joe' && birthDate[0] === '1955-11-05') {
         response.body = timestampBody(BILL_GARTON_SEARCHSET)
```
Comment (guyflavell2-nhs) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:154]: Is this correct?

Comment (guyflavell2-nhs) [reply karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:154]: perhaps rename or redefine HISTORIC_JOE_BLOGS to make it clear what is happening here?

Comment (PeterGallagherBJSS) [reply karate-tests/src/test/java/mocks/sandbox/get-patient-search.js]: The idea behind it is the request is has the fuzzy match flag set, and is matching on the historic name of Joe Blogs. 
I've renamed it to JOE_BLOGS_HISOTRIC_NAME_SEARCHSET.

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const JACKIE_JONES_SEARCHSET = context.read('classpath:mocks/stubs/searchRespons
 const RODNEY_GODSOE_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/rodney_godsoe_searchset.json')
 const MARTHA_MASSAM_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/martha_massam_searchset.json')
 const YOUDS_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/youds_searchset.json')
-const BILL_GARTON_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/bill_garton_searchset.json')
+const JOE_BLOGS_HISOTRIC_NAME_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/bill_garton_searchset.json')
```
Comment (ehsanhussain-nhsd) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:24]: File path still points to bill_garton_searchset.json

Comment (PeterGallagherBJSS) [reply karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:24]: It still needs to point to that record as Bill Garton is the current name / what it's returning. It's matching on Joe Blogs which is the historic name.
Or are you suggesting renaming the file as well to something like joe_blogs_historic_searchset.json?

**Review by ehsanhussain-nhsd — APPROVED** [review]

**Review by guyflavell2-nhs — APPROVED** [review]

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js (hunk 2 above)
Comment (claudiudumitrascu1-nhs) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:24]:
```suggestion
const JOE_BLOGS_HISTORIC_NAME_SEARCHSET = context.read('classpath:mocks/stubs/searchResponses/bill_garton_searchset.json')
```

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js [hunk 3]
```diff
@@ -149,14 +150,20 @@ if (request.pathMatches('/Patient') && request.get) {
 
   if (validateHeaders(request) && validateQueryParams(request)) {
     if (fuzzyMatch) {
-      if (family === 'Garton' && (given[0]) === 'Bill' && birthDate[0] === '1946-06-23') {
-        response.body = timestampBody(BILL_GARTON_SEARCHSET)
+      if (family === 'Blogs' && given[0] === 'Joe' && birthDate[0] === '1955-11-05') {
+        response.body = timestampBody(JOE_BLOGS_HISOTRIC_NAME_SEARCHSET)
```
Comment (claudiudumitrascu1-nhs) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:154]:
```suggestion
        response.body = timestampBody(JOE_BLOGS_HISTORIC_NAME_SEARCHSET)
```

File: karate-tests/src/test/java/mocks/sandbox/get-patient-search.js [hunk 4]
```diff
@@ -149,14 +150,20 @@ if (request.pathMatches('/Patient') && request.get) {
...
       } else if (phone === '01222111111' && email === 'test@test.com') {
         response.body = timestampBody(FUZZY_SINGLE_SEARCHSET)
       } else if (family === 'Smythe' && (given[0]) === 'Mat' && (birthDate[0]) === 'ge2000-05-03' && gender === 'male' && postalCode === 'DN17 4AA' && email !== 'rubbish@work.com') {
         response.body = timestampBody(FUZZY_MULTI_SEARCHSET)
+      } else if (['MED', 'HUME'].includes(family) && (given[0]) === 'Casey' && (birthDate[0]) === '1999-09-09') {
+        response.body = timestampBody(HISTORIC_DATA_SEARCHSET)
+      } else if (family === 'MED' && (given[0]) === 'Casey' && (birthDate[0]) === '2024-01-12') {
+        response.body = timestampBody(EMPTY_SEARCHSET)
+      } else if (family === 'LEEKE' && (given[0]) === 'Horace' && (birthDate[0]) === '1956-05-02' && postalCode === 'DN16') {
+        response.body = timestampBody(EMPTY_SEARCHSET)
```
Comment (claudiudumitrascu1-nhs) [inline karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:166]: Is there a better way to set the response.body, other than loads of "else if"s ? Perhaps some dictionary that we can loop through?

Comment (agelledi) [reply karate-tests/src/test/java/mocks/sandbox/get-patient-search.js:166]: I will create dedicated technical debt ticket and  refactor existing else if structures using switch or dictionary

**Review by claudiudumitrascu1-nhs — APPROVED** [review]

**Pull request merged**